subprojects/plugins/src/integTest/groovy/org/gradle/groovy/GroovyCrossCompilationIntegrationTest.groovy

```diff
@@ -19,84 +19,61 @@ package org.gradle.groovy
 import org.gradle.api.JavaVersion
 import org.gradle.integtests.fixtures.AvailableJavaHomes
 import org.gradle.integtests.fixtures.MultiVersionIntegrationSpec
-import org.gradle.integtests.fixtures.TargetVersions
-import org.gradle.internal.jvm.JavaInfo
-import org.gradle.test.fixtures.file.ClassFile
-import org.gradle.util.internal.TextUtil
+import org.gradle.integtests.fixtures.TargetCoverage
+import org.gradle.integtests.fixtures.jvm.JavaToolchainFixture
+import org.gradle.testing.fixture.GroovyCoverage
+import org.gradle.util.internal.VersionNumber
 import org.junit.Assume
 
-@TargetVersions(["1.6", "1.7", "1.8"])
-class GroovyCrossCompilationIntegrationTest extends MultiVersionIntegrationSpec {
-    JavaVersion getJavaVersion() {
-        JavaVersion.toVersion(MultiVersionIntegrationSpec.version)
-    }
+import static org.gradle.util.internal.GroovyDependencyUtil.groovyModuleDependency
 
-    JavaInfo getTarget() {
-        return AvailableJavaHomes.getJdk(javaVersion)
-    }
+@TargetCoverage({ GroovyCoverage.SINCE_3_0 })
+class GroovyCrossCompilationIntegrationTest extends MultiVersionIntegrationSpec implements JavaToolchainFixture {
 
-    def setup() {
-        Assume.assumeTrue(target != null)
-        def java = TextUtil.escapeString(target.getJavaExecutable())
-        def javaHome = TextUtil.escapeString(target.javaHome.absolutePath)
+    def "can compile source and run tests using Java #javaVersion for Groovy "() {
```

Comment:
❓ What is the new thing we test here? That we can compile for a JDK 6 or 7 target?

Reply:
Ah, I read the filter wrong. So this only runs tests on JDK >= 8. Does that mean we can't compile to JDK 6/7 for older Groovy versions any more? Would that work when only specifying source/target compatibility?

Reply:
We still can compile to older Java targets (such as 1.6) with Groovy pre-3 version. However, for that compilation must run in a worker with an older Java as well, such as Java 11.

I added `GroovyCompileOldJavaTargetIntegrationTest` test to capture this behavior.